spec.html

```diff
@@ -36582,7 +36582,7 @@ <h1>WeakMap.prototype.delete ( _key_ )</h1>
           1. Let _M_ be the *this* value.
           1. Perform ? RequireInternalSlot(_M_, [[WeakMapData]]).
           1. Let _entries_ be the List that is _M_.[[WeakMapData]].
-          1. If Type(_key_) is not Object, return *false*.
+          1. If Type(_key_) is not Object or Symbol, return *false*.
```

Comment:
not sure if this would be overkill, but it might be nice to have a single abstract operation that owns this definition instead of having to change it in 4-8 places

Reply:
Sure, I'm open to that. I guess the general question would be, do we want to define a term for this concept of being an Object or Symbol? For example, "referenceable" or "identity-full"? If someone can think of a real word describing this concept, that'd probably be better than those two...

Reply:
forgeable?

Reply:
forgeable means something different :-)

"weakable" sounds weird tho

Reply:
I would say that the general concept is something like "distinguishable".

Reply:
"distinguishable" might be a bit too broad, since 3 and 4 are distinguishable but depending on your mental model:
 - if each `3` is a different 3, but `3 === 3` compares them by value and not by identity/reference, then symbols and objects "have identity" but the rest don't.
 - if every `3` is the *same 3*, then you can create symbols and objects, but the rest are more like singletons referenced by syntax, in which case i'm not sure what you'd call it.

Reply:
We're going to be able to poke holes in whatever term we come up with; the key will be to socially agree on some shared language here. I could see using either "unforgeable" or "distinguishable" if we agreed on it, and adjusted our usages in other places appropriately.

Reply:
`HasIdentity`, perhaps? (It needn't be a single adjective.)

Reply:
What if we used a phrase, "has identity"? We could still define and linkify this.

Reply:
I'd prefer it as an abstract operation, personally, to match `IsCallable` and [friends](https://tc39.es/ecma262/#sec-testing-and-comparison-operations). Don't feel strongly though.